### PR TITLE
Fix G-API smart classroom demo

### PR DESCRIPTION
**Pull request by smirnov-alexey**
Demo is converted to G-API OV 2.0 backend which solves the problem with assumed NCHW layout vs NHWC which we get from the network (IRv11)

### CODE REVIEW

**Comment by smirnov-alexey** [comment]
@Wovchena @TolyaTalamanov @dmatveev

File: demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include <vector>
 
 #include <opencv2/gapi/infer/ie.hpp>
+#include <opencv2/gapi/infer/ov.hpp>
```
Comment (Wovchena) [inline demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp:15]: `Build Clang Demos (Ubuntu 20.04):` `/opt/home/sys_k8sworker/ci-ov-omz/workspace/omz/precommit/build_clang_demos_ubuntu20/src/demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp:15:10: fatal error: 'opencv2/gapi/infer/ov.hpp' file not found`

I believe upgrading demos to G-API with ov 2.0 was going to be a separate activity

**Comment by smirnov-alexey** [comment]
Should be green once OpenCV PR is merged

File: demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp (hunk 1 above)
Comment (smirnov-alexey) [reply demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp:15]: Discussed locally. I disabled G-API build with low OpenCV version. Also enabled some tests in github-actions according to https://github.com/openvinotoolkit/open_model_zoo/pull/3878

File: demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp [hunk 2]
```diff
@@ -188,26 +189,26 @@ void printInfo(const NetsFlagsPack& flags, std::string& teacher_id, std::string&
...
+        const std::vector<std::string> action_detector_6 = {"ActionNet/out_detection_loc",
+                                                            "ActionNet/out_detection_conf",
+                                                            "ActionNet/action_heads/out_head_1_anchor_1",
+                                                            "ActionNet/action_heads/out_head_2_anchor_1",
+                                                            "ActionNet/action_heads/out_head_2_anchor_2",
+                                                            "ActionNet/action_heads/out_head_2_anchor_3",
+                                                            "ActionNet/action_heads/out_head_2_anchor_4"};
         /** Create action detector net's parameters **/
-        std::array<std::string, 7> outputBlobList =
+        std::vector<std::string> outputBlobList =
```
Comment (Wovchena) [inline demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp:207]: Why vector now?

**Review by Wovchena — APPROVED** [review]

File: demos/tests/cases.py [hunk 3]
```diff
@@ -268,22 +268,21 @@ def single_option_cases(key, *args):
...
-    #         TestCase(options={'-m_a': ModelArg('common-sign-language-0002'),
-    #                           '-c': str(OMZ_DIR / 'data/dataset_classes/common_sign_language12.json')}),
-    #     ],
-    # )),
+    CppDemo(name='gesture_recognition_demo', implementation='cpp_gapi',
+            model_keys=['-m_a', '-m_d'],
+            device_keys=['-d_a', '-d_d'],
+            test_cases=combine_cases(
+        TestCase(options={'--no_show': None,
+                          '-i': TestDataArg('msasl/global_crops/_nz_sivss20/clip_0017/img_%05d.jpg'),
```
Comment (Wovchena) [inline demos/tests/cases.py:276]:
```suggestion
                          '-i': DataPatternArg('coco128-every-480x640x3'),
```

**Comment by Wovchena** [comment]
I suppose you are OK if this patch isn't included into 23.3 release because it conditionally disables compilation of all GAPI demos just to fix two of them. So I'm not going to rush updating openvino submodule, moreover demos aren't included in packages anyway, so this is more like a formal process.

**Comment by smirnov-alexey** [comment]
@Wovchena awesome, thank you so much! Sure, we can discuss 23.3 release later. Probably, we will convert other demos to OV 2.0 backend and will prepare additional changes to enable demo validation

**Pull request merged**